src/table/boot.rs

```diff
@@ -206,12 +206,14 @@ impl BootServices {
         (self.free_pages)(addr, count).into()
     }
 
-    /// Retrieves the size, in bytes, of the current memory map.
-    ///
-    /// A buffer of this size will be capable of holding the whole current memory map,
-    /// including padding. Note, however, that allocations will increase the size of the
-    /// memory map, therefore it is better to allocate some extra space.
-    pub fn memory_map_size(&self) -> usize {
+    /// Returns a tuple where the first element is the size of a memory descriptor,
+    /// and the second element is the size of the current memory map. 
```

Comment:
Just an idea, would it be clearer if we updated the API to return a `struct` with named fields instead of just a pair of `usize`s?

I'm basing this on the justification in https://stackoverflow.com/a/40070506/5723188:

> Another example: imagine you are writing an application that somehow works with a text file (compiler, text editor, ...). When you are talking about a region in the text file (for example, a search result), you want to specify that region by giving byte offsets. Let's see if tuples work out for us:
> ```rust
> fn find_first_occurence(file: &TextFile, needle: &str) -> (usize, usize)
> ```
> Does the return type speak for itself? Rather not... even if you know that regions in the file are specified by byte offsets, the return value is still ambiguous: either it's (start, end) or it's (start, stuff), where stuff can be any other search metric (the function doesn't need to return end, since we already know the length of needle and thus could calculate it). So, I hope you agree, we want to name the return type. Let's call it Span -- that's the name used in the Rust compiler.

Something like `MemoryMapSize { entry_size: usize, map_size: usize }` would seem clearer.

Reply:
Just made the change, and also updated the tests as well.